packages/react-app/src/App.jsx

```diff
@@ -512,6 +512,14 @@ function App(props) {
               Support
             </Button>
           </Col>
+          {localProvider._network && localProvider._network.chainId == 31337 && (
```

Comment:
There's a proliferation of the checks for the hardhat chain in local network all over App.jsx.

I would suggest this pattern should be modified and there should a variable on top of app.jsx that replaces all these checks to make it easier for new devs to understand what is happening.  
```
const isHardHatNetwork = localProvider._network && localProvider._network.chainId == 31337
```
and replacing the 4-6 instances of the check in this file with that constant